src/lib/phdi-building-blocks/phdi_building_blocks/azure_blob.py

```diff
@@ -0,0 +1,30 @@
+import json
+import pathlib
+from azure.identity import DefaultAzureCredential
+from azure.storage.blob import ContainerClient
+
+
+def get_blob_client(container_url: str) -> ContainerClient:
+    """Use whatever creds Azure can find to authenticate with the storage container"""
+    creds = DefaultAzureCredential()
+    return ContainerClient.from_container_url(container_url, credential=creds)
+
+
+def store_data(
+    container_url: str,
+    prefix: str,
+    filename: str,
+    bundle_type: str,
+    message_json: dict = None,
+    message: str = None,
+) -> None:
+    """
+    Store the given data, which is either a FHIR bundle or an HL7 message in the
+    appropriate output container
+    """
+    client = get_blob_client(container_url)
+    blob = client.get_blob_client(str(pathlib.Path(prefix) / bundle_type / filename))
+    if message_json is not None:
+        blob.upload_blob(json.dumps(message_json).encode("utf-8"), overwrite=True)
```

Comment:
How will this interact with things once the pipeline has already been run once and we just want to process new data? Shouldn't we set `overwrite=False`) and not write the contents if a file of the same name already exists?

Reply:
This is existing code (just moved to another file).  overwrite=False was the original behavior, and was causing problems when we re-ran files after code updates and wanted those changes to simply overwrite the previous run's data.  For that reason, Graham had suggested we set overwrite=True which seemed like a good idea to me at the time.  It's not the most intelligent solution, and could be improved, but in any case, it's probably out of scope for this PR.

Reply:
That's probably a good catch. @spencer-skylight I believe you wrote this initially what do you think?

Reply:
That's sound reasoning, Spencer. This changeset is big enough as is so probably no need to modify the scope here, especially if things were causing problems the other way.